### DISCUSSION TITLE
Welcome to Tomotroid Discussions!

### DISCUSSION THREAD

**Discussion by Vadoola** [Vadoola/Tomotroid#58] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Risuleia** [comment]
Hi there!
So I just came upon this issue recently while trying to implement notifications via the `notify-rust` crate.
The notifications get sent from "Windows PowerShell" in Windows 11. The `appname` and `icon` functions in the `Notification` struct don't seem to be working on Windows 11.
Any ideas regarding that?

**Reply by Vadoola** [reply]
I have noticed that same issue, unfortunately I havent spent much time looking into why.

I assume it might have to do with the way the notify_rust is launching the notification on Windows. Perhaps shelling out to a power shell command instead of using the Windows APInfir notifications.

If I get some time this week I'll try and take a closer look, I imagine its an upstream issue that would need tonfet fixed in notify-rust though.

**Reply by Vadoola** [reply]
So I did a quick glance. It looks like there is code to handle an icon, so I'm not sure why that isn't working.

The App name, it looks like it might try lookup an App_ID (https://learn.microsoft.com/en-us/windows/win32/shell/appids) and if it fails, it defaults to using the Powershell AppID (probably because it exists on all modern Windows systems). I would need to look deeper to see what's involved in getting an AppID it looks like it might be something taken from the registry? I'm assuming generated when an App is installed?

**Reply by Risuleia** [reply]
> I have noticed that same issue, unfortunately I havent spent much time looking into why.
> 
> I assume it might have to do with the way the notify_rust is launching the notification on Windows. Perhaps shelling out to a power shell command instead of using the Windows APInfir notifications.
> 
> If I get some time this week I'll try and take a closer look, I imagine its an upstream issue that would need tonfet fixed in notify-rust though.

Mhm, it probably is an upstream issue. It defaults to the PowerShell AppID (so it can be backwards-compatible) when it can't find the one for the rust application.

**Reply by Vadoola** [reply]
Yeah. I'm not super familiar with Windows AppIDs but I guess, I'll need to dig into it so that the notifications show correctly.
